### PR TITLE
Feature/let user change default lang

### LINKED ISSUE #389: Let a user change its own default language
## Motivations
default_language is fetch from the login-module at first login/sign-in. Then, the user can choose among the languages supported in the platform.

## Subtasks
* On the personal page, move the language above "Personal Information".
* Change the label to "Platform language".
* Replace the value by a dropdown list including all the supported languages (check/reuse language picker)
* On language change, [update user's profile](https://france-ioi.github.io/algorea-devdoc/api/#operation/userDataUpdate) and redirect to the same page in the new language (improve LocaleService)

### PR DESCRIPTION
**Pull request by Iloveall**
Fixes #389

### CODE REVIEW

File: src/app/core/components/language-picker/language-picker.component.scss [hunk 1]
```diff
@@ -1,5 +1,15 @@
 @import "src/colors.scss";
 
+:host {
+  &.dark {
+    ::ng-deep .p-dropdown {
+      .p-dropdown-label {
+        color: $base-color;
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/core/components/language-picker/language-picker.component.scss]: Fixed

[the remark this answers is not in the document]
Comment (smadbe) [reply src/app/core/components/language-picker/language-picker.component.scss]: Screenshot 2021-05-12 at 10 26 52

The theme is still dark in the profile.

[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/core/components/language-picker/language-picker.component.scss]: Fixed

**Review by smadbe — APPROVED** [review]
👍

**Pull request merged**